deps/rabbit/src/rabbit_policy.erl

```diff
@@ -441,72 +430,69 @@ notify_clear(VHost, <<"operator_policy">>, Name, ActingUser) ->
 %% [2] We could be here in a post-tx fun after the vhost has been
 %% deleted; in which case it's fine to do nothing.
 update_matched_objects(VHost, PolicyDef, ActingUser) ->
-    Tabs = [rabbit_queue,    rabbit_durable_queue,
-            rabbit_exchange, rabbit_durable_exchange],
-    {XUpdateResults, QUpdateResults} = rabbit_misc:execute_mnesia_transaction(
-        fun() ->
-            _ = [mnesia:lock({table, T}, write) || T <- Tabs], %% [1]
-            case catch {list(VHost), list_op(VHost)} of
-                {'EXIT', {throw, {error, {no_such_vhost, _}}}} ->
-                    {[], []}; %% [2]
-                {'EXIT', Exit} ->
-                    exit(Exit);
-                {Policies, OpPolicies} ->
-                    {[update_exchange(X, Policies, OpPolicies) ||
-                        X <- rabbit_exchange:list(VHost)],
-                    [update_queue(Q, Policies, OpPolicies) ||
-                        Q <- rabbit_amqqueue:list(VHost)]}
-                end
-        end),
+    {XUpdateResults, QUpdateResults} =
+        case catch {list(VHost), list_op(VHost)} of
+            {'EXIT', {throw, {error, {no_such_vhost, _}}}} ->
+                {[], []}; %% [2]
+            {'EXIT', Exit} ->
+                exit(Exit);
+            {Policies, OpPolicies} ->
+                rabbit_db_policy:update(VHost,
+                                        get_updated_exchange(Policies, OpPolicies),
+                                        get_updated_queue(Policies, OpPolicies))
+        end,
```

Comment:
Listing the policies for `VHost` is outside the transaction. I have the feeling we already talked about this in Slack, but I can't find the discussion.

Is it ok to list policies outside of the transaction?

Reply:
I think it is. Each policy we set generates one of this `notify` callbacks, so let's say that we update one policy and one operator policy more or less simultaneously. If the first update doesn't pick up both policies, the second one will. Also 
 the same for clear notifications. We don't have an atomic way to update policies, so if there is a temporary mismatch I don't think it can be an issue.

Reply:
Perfect, thank you!